File: backend/app/schemas/meeting.py
from pydantic import BaseModel, Field
from datetime import datetime
from typing import Optional, List
from app.schemas.participant import ParticipantResponse


class MeetingBase(BaseModel):
    title: str = Field(..., min_length=1, max_length=200)
    description: Optional[str] = None
    host_name: str = Field(..., min_length=1, max_length=100)
    scheduled_date: Optional[datetime] = None
    scheduled_time: Optional[str] = None
    duration: Optional[int] = None


class MeetingCreate(MeetingBase):
    pass


class MeetingSchedule(MeetingBase):
    scheduled_date: datetime = Field(...)
    scheduled_time: str = Field(..., min_length=1, max_length=10)
    duration: int = Field(..., gt=0)


class MeetingJoin(BaseModel):
    meeting_id: str = Field(..., min_length=1)
    display_name: str = Field(..., min_length=1, max_length=100)


class MeetingUpdate(BaseModel):
    title: Optional[str] = Field(None, min_length=1, max_length=200)
    description: Optional[str] = None
    scheduled_date: Optional[datetime] = None
    scheduled_time: Optional[str] = None
    duration: Optional[int] = None
    is_active: Optional[bool] = None


class MeetingResponse(MeetingBase):
    id: int
    meeting_id: str
    is_active: bool
    created_at: datetime
    updated_at: Optional[datetime] = None
    participants: List[ParticipantResponse] = []

    class Config:
        from_attributes = True


class MeetingListResponse(BaseModel):
    id: int
    meeting_id: str
    title: str
    host_name: str
    scheduled_date: Optional[datetime] = None
    scheduled_time: Optional[str] = None
    is_active: bool
    created_at: datetime

    class Config:
        from_attributes = True

File: frontend/src/types/meeting.ts
export interface Participant {
  id: number
  meeting_id: number
  display_name: string
  is_muted: boolean
  is_camera_on: boolean
  joined_at: string
}

export interface Meeting {
  id: number
  meeting_id: string
  title: string
  description?: string
  host_name: string
  scheduled_date?: string
  scheduled_time?: string
  duration?: number
  is_active: boolean
  created_at: string
  updated_at?: string
  participants: Participant[]
}

export interface MeetingList {
  id: number
  meeting_id: string
  title: string
  host_name: string
  scheduled_date?: string
  scheduled_time?: string
  is_active: boolean
  created_at: string
}

export interface MeetingCreate {
  title: string
  description?: string
  host_name: string
  scheduled_date?: string
  scheduled_time?: string
  duration?: number
}

export interface MeetingSchedule {
  title: string
  description?: string
  host_name: string
  scheduled_date: string
  scheduled_time: string
  duration: number
}

export interface MeetingJoin {
  meeting_id: string
  display_name: string
}

export interface MeetingUpdate {
  title?: string
  description?: string
  scheduled_date?: string
  scheduled_time?: string
  duration?: number
  is_active?: boolean
}
